### PR TITLE
fix: add numbers localization

### PR DESCRIPTION
**Pull request by MonicaTanios**
Co-authored-by: Micheline Medhat <michelinemedhat@gmail.com>

### CODE REVIEW

File: lib/l10n/tr_nums.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import 'dart:io';
+import 'package:intl/intl.dart';
+
+String trNums(dynamic num) {
+  final formatter = NumberFormat('###.0#', Platform.localeName);
```
Comment (MichaelSafwatHanna) [inline lib/l10n/tr_nums.dart]:
```suggestion
String trNumber(num number) {
  final formatter = NumberFormat('###.#', Platform.localeName);
```

use `num` type to force passing numbers to the function and the pattern change to remove the zero after decimal if there's no mantissa

File: lib/l10n/tr_nums.dart [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import 'package:intl/intl.dart';
+
+String trNums(num number) {
+  final formatter = NumberFormat('###.#', Platform.localeName);
+  return formatter.format(num.parse(format(number)));
+}
+
+String format(num n) {
+  return n.toStringAsFixed(n.truncateToDouble() == n ? 0 : 2);
+}
```
Comment (yousinix) [inline lib/l10n/tr_nums.dart]: Again, we don't need this

From [NumberFormat](https://api.flutter.dev/flutter/intl/NumberFormat-class.html) docs:
> `#` A single digit, omitted if the value is zero

File: lib/l10n/tr_nums.dart [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import 'dart:io';
+import 'package:intl/intl.dart';
+
+String trNums(num number) {
+  final formatter = NumberFormat('###.#', Platform.localeName);
```
Comment (yousinix) [inline lib/l10n/tr_nums.dart]: Get locale from `context` not from `Platform`
Note that `context` should be the first parameter, following flutter's convention.

**Review by yousinix — CHANGES_REQUESTED** [review]

File: lib/l10n/tr_nums.dart [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import 'dart:io';
+import 'package:intl/intl.dart';
+
+String trNums(num number) {
```
Comment (yousinix) [inline lib/l10n/tr_nums.dart]: Rename to `trNum` or `trNumber` (not plural) and move to the `tr.dart` file

**Review by yousinix — CHANGES_REQUESTED** [review]

File: lib/l10n/tr_nums.dart (hunk 3 above)
Comment (MichelineMedhat) [reply lib/l10n/tr_nums.dart]: context returns 'ar' while we need 'ar_EG', [country code is missing].

**Review by yousinix — APPROVED** [review]

**Review by MichaelSafwatHanna — APPROVED** [review]

**Pull request merged**